keadm/cmd/keadm/app/cmd/cloud/manifest.go

```diff
@@ -101,8 +101,8 @@ func addManifestsGenerateJoinOtherFlags(cmd *cobra.Command, initOpts *types.Init
 		"Skip printing the contents of CRDs to stdout")
 }
 
-//AddManifestsGenerate2ToolsList Reads the flagData (containing val and default val) and join options to fill the list of tools.
-func AddManifestsGenerate2ToolsList(toolList map[string]types.ToolsInstaller, flagData map[string]types.FlagData, initOpts *types.InitOptions) error {
+// AddManifestsGenerate2ToolsList Reads the flagData (containing val and default val) and join options to fill the list of tools.
```

Comment:
Thanks for the fixing, but I think the flagData still need to generate the manifest, you can explore how to use it :) Thanks @yeqiugt